go/externals/proof_service_generic_social.go

```diff
@@ -0,0 +1,158 @@
+		return nil, err
+	}
+	return gsConfig, nil
+}
+
+func (c *GenericSocialProofConfig) validate() error {
+	re, err := regexp.Compile(c.Username.Re)
+	if err != nil {
+		return err
+	}
+	c.usernameRe = re
+	if !strings.Contains(c.PrefillUrl, kbUsernameKey) {
+		return fmt.Errorf("invalid PrefillUrl %s, missing %s", c.PrefillUrl, kbUsernameKey)
+	}
+	if !strings.Contains(c.PrefillUrl, sigHashKey) {
+		return fmt.Errorf("invalid PrefillUrl %s, missing %s", c.PrefillUrl, sigHashKey)
+	}
+	if !strings.Contains(c.CheckUrl, usernameKey) {
+		return fmt.Errorf("invalid CheckUrl %s, missing %s", c.CheckUrl, usernameKey)
+	}
+	return nil
+}
+
+func (c *GenericSocialProofConfig) prefillURLWithValues(kbUsername string, sigID keybase1.SigID) (string, error) {
+	url := strings.Replace(c.PrefillUrl, kbUsernameKey, kbUsername, 1)
+	url = strings.Replace(url, sigHashKey, sigID.String(), 1)
+	if !strings.Contains(url, kbUsername) {
+		return "", fmt.Errorf("Invalid PrefillUrl, missing kbUsername")
+	}
+	if !strings.Contains(url, sigID.String()) {
+		return "", fmt.Errorf("Invalid PrefillUrl, missing sigHash")
+	}
+	return url, nil
+}
+
+//=============================================================================
+
+type GenericSocialProofServiceType struct {
+	libkb.BaseServiceType
+	config *GenericSocialProofConfig
+}
+
+func NewGenericSocialProofServiceType(config *GenericSocialProofConfig) GenericSocialProofServiceType {
+	return GenericSocialProofServiceType{
+		config: config,
+	}
+}
+
+func (t GenericSocialProofServiceType) AllStringKeys() []string { return t.BaseAllStringKeys(t) }
+
+func (t GenericSocialProofServiceType) NormalizeUsername(s string) (string, error) {
+	if !t.config.usernameRe.MatchString(s) {
+		return "", libkb.NewBadUsernameError(s)
+	}
+	// TODO always normalize ToLower?
+	return strings.ToLower(s), nil
+}
+
+func (t GenericSocialProofServiceType) NormalizeRemoteName(m libkb.MetaContext, s string) (ret string, err error) {
+	return t.NormalizeUsername(s)
+}
+
+func (t GenericSocialProofServiceType) GetPrompt() string {
+	return fmt.Sprintf("Your username on %s", t.config.DisplayName)
+}
+
+func (t GenericSocialProofServiceType) ToServiceJSON(username string) *jsonw.Wrapper {
+	return t.BaseToServiceJSON(t, username)
+}
+
+func (t GenericSocialProofServiceType) PostInstructions(username string) *libkb.Markup {
+	return libkb.FmtMarkup(`Please click on the following link to post to %v:`, t.config.DisplayName)
+}
+
+func (t GenericSocialProofServiceType) DisplayName(username string) string {
+	return t.config.DisplayName
+}
+func (t GenericSocialProofServiceType) GetTypeName() string { return t.config.Domain }
+
+func (t GenericSocialProofServiceType) RecheckProofPosting(tryNumber int, status keybase1.ProofStatus, _ string) (warning *libkb.Markup, err error) {
+	return t.BaseRecheckProofPosting(tryNumber, status)
+}
+
+func (t GenericSocialProofServiceType) GetProofType() string {
+	return libkb.GenericSocialWebServiceBinding
```

Comment:
Does this work for all the callers? It's a bit of a departure from how the other social proofs work.
Example: https://github.com/keybase/client/blob/1d3f53ed7acada1222991909a84f15da27975323/go/libkb/identify_outcome.go#L80
Several other callers are RPC exports

Reply:
there is some cleanup around this but will address it in the next PR since that's when the `GENERIC_SOCIAL` type is introduced: https://github.com/keybase/client/pull/13846/files#diff-ef8fd59916536fb8e588f2c9bc209c4cR84

Reply:
ok